sklearn/metrics/_ranking.py

```diff
@@ -1441,3 +1441,151 @@ def ndcg_score(y_true, y_score, *, k=None, sample_weight=None,
     _check_dcg_target_type(y_true)
     gain = _ndcg_sample_scores(y_true, y_score, k=k, ignore_ties=ignore_ties)
     return np.average(gain, weights=sample_weight)
+
+
+def top_k_accuracy_score(y_true, y_score, *, k=2, normalize=True,
+                         sample_weight=None, labels=None):
+    """Top-k Accuracy classification score.
+
+    This metric computes the number of times where the correct label is among
+    the top `k` labels predicted (ranked by predicted scores). Note that the
+    multilabel case isn't covered here.
+
+    Read more in the :ref:`User Guide <top_k_accuracy_score>`
+
+    Parameters
+    ----------
+    y_true : array-like of shape (n_samples,)
+        True labels.
+
+    y_score : array-like of shape (n_samples,) or (n_samples, n_classes)
+        Target scores. These can be either probability estimates or
+        non-thresholded decision values (as returned by
+        :term:`decision_function` on some classifiers). The binary case expects
+        scores with shape (n_samples,) while the multiclass case expects scores
+        with shape (n_samples, n_classes). In the nulticlass case, the order of
+        the class scores must correspond to the order of ``labels``, if
+        provided, or else to the numerical or lexicographical order of the
+        labels in ``y_true``.
+
+    k : int, default=2
+        Number of guesses allowed to find the correct label.
```

Comment:
I'm not sure the the word "guess" is appropriate here. It implies sampling, while this approach is fully deterministic. Maybe

```suggestion
        Number of most likely outcomes considered to find the
        correct label.
```